src/Config/HostsFile.php

```diff
@@ -0,0 +1,146 @@
+    }
+
+    /**
+     * Loads a hosts file (from the given path or default location)
+     *
+     * Note that this method blocks while loading the given path and should
+     * thus be used with care! While this should be relatively fast for normal
+     * hosts file, this may be an issue if this file is located on a slow device
+     * or contains an excessive number of entries. In particular, this method
+     * should only be executed before the loop starts, not while it is running.
+     *
+     * @param ?string $path (optional) path to hosts file or null=load default location
+     * @return self
+     * @throws RuntimeException if the path can not be loaded (does not exist)
+     */
+    public static function loadFromPathBlocking($path = null)
+    {
+        if ($path === null) {
+            $path = self::getDefaultPath();
+        }
+
+        $contents = @file_get_contents($path);
+        if ($contents === false) {
+            throw new RuntimeException('Unable to load hosts file "' . $path . '"');
+        }
+
+        return new self($contents);
+    }
+
+    /**
+     * Instantiate new hosts file with the given hosts file contents
+     *
+     * @param string $contents
+     */
+    public function __construct($contents)
+    {
+        // remove all comments from the contents
+        $contents = preg_replace('/ *#.*/', '', strtolower($contents));
+
+        $this->contents = $contents;
+    }
+
+    /**
+     * Returns all IPs for the given hostname
+     *
+     * @param string $name
+     * @return string[]
+     */
+    public function getIpsForHost($name)
+    {
+        $name = strtolower($name);
+
+        $ips = array();
+        foreach (preg_split('/\r?\n/', $this->contents) as $line) {
+            $parts = preg_split('/\s+/', $line);
+            $ip = array_shift($parts);
+            if ($parts && array_search($name, $parts) !== false) {
+                // remove IPv6 zone ID (`fe80::1%lo0` => `fe80:1`)
+                if (strpos($ip, ':') !== false && ($pos = strpos($ip, '%')) !== false) {
+                    $ip= substr($ip, 0, $pos);
+                }
+
+                $ips[] = $ip;
+            }
+        }
+
+        return $ips;
+    }
+
+    /**
+     * Returns all hostnames for the given IPv4 or IPv6 address
+     *
+     * @param string $ip
+     * @return string[]
+     */
+    public function getHostsForIp($ip)
+    {
+        // check binary representation of IP to avoid string case and short notation
+        $ip = @inet_pton($ip);
+
+        $names = array();
+        foreach (preg_split('/\r?\n/', $this->contents) as $line) {
+            $parts = preg_split('/\s+/', $line);
+            $addr = array_shift($parts);
+
+            // remove IPv6 zone ID (`fe80::1%lo0` => `fe80:1`)
+            if (strpos($addr, ':') !== false && ($pos = strpos($addr, '%')) !== false) {
+                $addr = substr($addr, 0, $pos);
+            }
+
+            if (@inet_pton($addr) === $ip) {
```

Comment:
I guess it should just error out if `$ip = @inet_pton($ip)` returns `false`, might help with programming errors.